File: clinical-mdr-api/clinical_mdr_api/domains/study_selections/study_design_cell.py
import datetime
from dataclasses import dataclass


@dataclass
class StudyDesignCellVO:
    study_uid: str
    study_epoch_uid: str
    study_element_uid: str
    transition_rule: str
    order: int

    # Study selection Versioning
    start_date: datetime.datetime
    author_id: str | None
    author_username: str | None = None

    uid: str | None = None
    study_element_name: str | None = None
    study_epoch_name: str | None = None
    study_arm_uid: str | None = None
    study_arm_name: str | None = None
    study_branch_arm_uid: str | None = None
    study_branch_arm_name: str | None = None

    def edit_core_properties(
        self,
        order: int,
        transition_rule: str,
        study_epoch_uid: str,
        study_element_uid: str,
        study_arm_uid: str | None,
        study_branch_arm_uid: str | None,
    ):
        self.study_epoch_uid = study_epoch_uid
        self.study_element_uid = study_element_uid
        self.study_arm_uid = study_arm_uid
        self.study_branch_arm_uid = study_branch_arm_uid
        self.transition_rule = transition_rule
        self.order = order

File: studybuilder-ddf-api-adaptor/ddf-translator-lib/src/main/java/org/openstudybuilder/model/StudyDesignCell.java
package org.openstudybuilder.model;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.Getter;
import lombok.Setter;

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@Getter @Setter
public class StudyDesignCell {
    private String studyUid;
    private String designCellUid;
    private String studyArmUid;
    private String studyArmName;
    private String studyBranchArmName;
    private String studyBranchArmUid;
    private String studyEpochName;
    private String studyEpochUid;
    private String studyElementName;
    private String studyElementUid;
    private String transitionRule;
    private String startDate;
    private String userInitials;
    private String endDate;
    private int order;
}
